frontend/src/scenes/dashboard/dashboardLogic.tsx

```diff
@@ -773,7 +827,12 @@ export const dashboardLogic = kea<dashboardLogicType>({
             }
             const dashboardId: number = props.id
 
-            const insights = tiles?.map((t) => t.insight) || values.items || []
+            const insights = (tiles || values.insightTiles || []).reduce((acc, cur) => {
+                if (cur.insight) {
+                    acc.push(cur.insight)
+                }
+                return acc
+            }, [] as InsightModel[])
```

Comment:
I'm still 100% convinced that `reduce` makes for fugly code. It *always* raises my "someone's writing clever code, not clean code" detector. The map/filter version is so much cleaner and easier to follow, no?

```ts
const insights = (tiles || values.insightTiles || []).filter(t => !!t.insight).map(t => t.insight)
```

Also, if you ever have to write `as Type`, I'm also going to comment that this loses type strictness and might hide bugs 🙈 

Might be worth upgrading TS itself and using `satisfies` instead?

Reply:
> The map/filter version is so much cleaner and easier to follow, no?

I see reduce, I think: "ah many things are becoming one"

I don't find it any harder to follow than anything else (in and of itself). It's just a part of the language

Reply:
```
      const insights = (tiles || values.insightTiles || [])
          .map((t) => t.insight)
          .filter((i): i is InsightModel => !!i)
```

ended up with this since otherwise TS doesn't know the type is `InsightModel` and not `InsightModel | undefined`